src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,77 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
+const ERROR_MESSAGE = 'Failed to load users, try again later';
+
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [user, setUser] = useState<User | null>(null);
+
+  const handleCloseButtonClick = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .catch(() => setError(ERROR_MESSAGE))
+        .finally(() => setIsLoading(false));
+    }
+  }, []);
```

Comment:
one more:
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/86923089/2e41f401-9e09-4e9b-b3a9-fee98d19b067)